app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,11 @@
+# Controller for user features not provided by Devise - profile pages, etc
+class Users::SessionsController < Devise::SessionsController
+  def new
```

Comment:
It looks like other `Devise::SessionsController` actions (e.g. `create`) should prevent logins too. If yes, consider using a `before_action` to protect multiple actions/methods in one go.